### PR TITLE
TOOLS-2502 TOOLS-2509 TOOLS-2510 Improve asadm's e2e tests

### PR DESCRIPTION
**Pull request by jdogmcsteezy**
Most of the work involves adding e2e tests for better coverage. While creating the tests I ran into a couple of problems that were easy enough to fix. One enhancement was to allow collectinfo mode to display node names (fqdn) and not just IP addresses in tables. See https://aerospike.atlassian.net/browse/TOOLS-2509. Another enhancement (or bug) was to keep the `show latencies` cmd from showing `~~` when a node returned empty histograms.

### CODE REVIEW

File: lib/utils/util.py [hunk 1]
```diff
@@ -133,23 +135,31 @@ def capture_stdout_and_stderr(func, *args, **kwargs):
...
+
+    stdout_capturer.flush()
+    stderr_capturer.flush()
+
+    return stdout_capturer.getvalue(), stderr_capturer.getvalue()
+
     sys.stdout.flush()
     stdout_old = sys.stdout
-    stdout_capturer = io.StringIO()
     sys.stdout = stdout_capturer
```
Comment (dwelch-spike) [inline lib/utils/util.py:153]: Need the commented code? Looks like stdout isn't being captured?

Is the code after the return on line 148 reachable? If it isn't it should be removed.

**Review by dwelch-spike — CHANGES_REQUESTED** [review]
Commented on some things that look like they could be cleaned up. After these are resoled it will LGTM.

File: lib/collectinfo_analyzer/collectinfo_handler/collectinfo_log.py [hunk 2]
```diff
@@ -147,12 +147,15 @@ def get_node_displaynames(self, nodes=None):
...
+                node_names = self.get_data(type="meta_data", stanza="node_names")
+                # "node_name" was stored in collectinfo file in asadm 2.15.0
+                if not node_names:
+                    node_names = self.cinfo_data.keys()
             else:
                 return {}
 
             for node_name in node_names:
-                self.node_names[node_name] = node_name
+                self.node_names[node_name] = node_names[node_name]
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/collectinfo_analyzer/collectinfo_handler/collectinfo_log.py:158]: `node_names` is actually a dict like {ip: node_name}. Before the changes in the PR it was {ip:ip} so either worked.

[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/collectinfo_analyzer/collectinfo_handler/collectinfo_log.py:158]: Ill change the name of `node_name`

File: lib/collectinfo_analyzer/collectinfo_handler/collectinfo_log.py [hunk 3]
```diff
@@ -398,7 +401,7 @@ def get_cluster_name(self):
 
     def _set_nodes(self, nodes):
         for node in nodes:
-            self.node_names[node] = node
+            self.node_names[node] = nodes[node]
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/collectinfo_analyzer/collectinfo_handler/collectinfo_log.py:404]: No, `node` is an ip address. `nodes` used to be a dict like {ip:ip} and it is now {ip: node_name}.

File: lib/collectinfo_analyzer/collectinfo_handler/collectinfo_log.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/collectinfo_analyzer/collectinfo_handler/collectinfo_log.py:158]: [This ticket](https://aerospike.atlassian.net/browse/TOOLS-2509) explains the change with an example. It should help clear up the change

File: lib/utils/util.py (hunk 1 above)
Comment (jdogmcsteezy) [reply lib/utils/util.py:153]: yeah, ill fix this. I was testing it out. This code is only used in tests as far as I can tell. I could not get stderr capturing working with `pytest` for some reason. I have waisted a lot of time trying to get this to work. A couple weeks back and a couple years ago I think. It works find with `unittest`

File: lib/utils/util.py [hunk 4]
```diff
@@ -133,23 +135,31 @@ def capture_stdout_and_stderr(func, *args, **kwargs):
...
-    stderr_capturer = io.StringIO()
     sys.stderr = stderr_capturer
 
-    await func(*args, **kwargs)
-
     stdout_output = stdout_capturer.getvalue()
     sys.stdout = stdout_old
 
     stderr_output = stderr_capturer.getvalue()
+    print("test", stderr_output)
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/utils/util.py:162]: yes

File: tox.ini [hunk 5]
```diff
@@ -1,6 +1,11 @@
 [coverage:run]
-include = 
-    lib/*
+# concurrency = multiprocessing
+disable_warnings = true
+parallel = true
+# source = /Users/jesseschmidt/Developer/aerospike-admin/
+
+[coverage:report]
+include = lib/*, asadm.py
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply tox.ini:8]: not needed anymore

File: lib/live_cluster/show_controller.py [hunk 6]
```diff
@@ -451,19 +451,24 @@ async def do_dc(self, line):
...
+                # self.view.show_config,
+                # "%s DC Configuration" % (dc),
+                # configs,
+                # self.cluster,
+                # title_every_nth=title_every_nth,
+                # flip_output=flip_output,
+                # **self.mods,
             )
-            for dc, configs in dc_configs.items()
+            # for dc, configs in dc_configs.items()
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/live_cluster/show_controller.py:471]: yes

File: lib/utils/util.py (hunk 1 above)
Comment (jdogmcsteezy) [reply lib/utils/util.py:153]: I decided to remove this function

**Review by dwelch-spike — APPROVED** [review]
LGTM

**Pull request merged**